core/dbt/events/types.py

```diff
@@ -2161,6 +2178,9 @@ def message(self) -> str:
 @dataclass
 class NodeStart(DebugLevel, Cli, File):
     unique_id: str
+    report_node_data: ParsedModelNode
+    node_status: str = node_statuses['running']
+    node_state: str = node_states['started']
```

Comment:
How would you feel about making a new ABC superclass that's below `Event` that communicates that this, and other, concrete events must have these three fields? That way we can have mypy enforce these fields on some but not others which lets us get rid of them as Optional on `Event`.

Reply:
Makes sense.